pylate/losses/cached_contrastive.py

```diff
@@ -0,0 +1,348 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from contextlib import nullcontext
+from functools import partial
+from typing import Iterable, Optional
+
+import torch
+import torch.nn.functional as F
+import tqdm
+from torch import Tensor, nn
+from torch.utils.checkpoint import get_device_states, set_device_states
+
+from ..models import ColBERT
+from ..scores import colbert_scores
+from .contrastive import extract_skiplist_mask
+
+
+class RandContext:
+    """
```

Comment:
Description should start on the same line as the quote """ for documentation parsing